Services/UI/OJS.Services.Ui.Business/Implementations/SubmissionsBusinessService.cs

```diff
@@ -579,16 +554,51 @@ public async Task<PagedResult<SubmissionForPublicSubmissionsServiceModel>> GetBy
             .ToPagedResultAsync(DefaultSubmissionsPerPage, page);
     }
 
-    public async Task<PagedResult<SubmissionForPublicSubmissionsServiceModel>> GetPendingSubmissions(int page)
-        => await this.submissionsCommonData
-            .GetAllPending()
-            .OrderByDescending(s => s.Id)
-            .MapCollection<SubmissionForPublicSubmissionsServiceModel>()
-            .ToPagedResultAsync(DefaultSubmissionsPerPage, page);
-
     public Task<int> GetTotalCount()
         => this.submissionsData.GetTotalSubmissionsCount();
 
+    public async Task<PagedResult<SubmissionForPublicSubmissionsServiceModel>> GetSubmissions(SubmissionEnumType type, int page)
+    {
+        if (type == SubmissionEnumType.Public)
+        {
+            var user = this.userProviderService.GetCurrentUser();
+
+            if (user.IsAdminOrLecturer)
+            {
+                return await this.submissionsData.GetLatestSubmissions<SubmissionForPublicSubmissionsServiceModel>(
+                    DefaultSubmissionsPerPage, page);
+            }
+
+            var modelResult = new PagedResult<SubmissionForPublicSubmissionsServiceModel>();
+
+            modelResult.Items =
+                await this.submissionsData.GetLatestSubmissions<SubmissionForPublicSubmissionsServiceModel>(
+                    DefaultSubmissionsPerPage);
+
+            return modelResult;
+        }
+
+        if (type == SubmissionEnumType.Processing)
+        {
+            return await this.submissionsCommonData
+                .GetAllProcessing()
+                .OrderByDescending(s => s.Id)
+                .MapCollection<SubmissionForPublicSubmissionsServiceModel>()
+                .ToPagedResultAsync(DefaultSubmissionsPerPage, page);
+        }
+
+        if (type == SubmissionEnumType.Pending)
+        {
+            return await this.submissionsCommonData
+                .GetAllPending()
+                .OrderByDescending(s => s.Id)
+                .MapCollection<SubmissionForPublicSubmissionsServiceModel>()
+                .ToPagedResultAsync(DefaultSubmissionsPerPage, page);
+        }
+
+        return null!;
```

Comment:
There shouldnt be a case where we return null. The return of `All` submissions should be the default behavior.